### PR TITLE
Prevent cache collisions in multisite install tasks

### LINKED ISSUE #3158: Refactor PR 3006 to Unblock Deploys/Updates
This is a parent issue to track progress of breaking a previously unmerged PR #3006 into smaller PRs to facilitate faster review. Proposed breakdown: 

- [x] Create script that creates and sets unique drush cache directory per site process for use by long running drush tasks
- [x] Refactor DRUSH_PATHS_CACHE_DIRECTORY approach
- [x] Enhance post-install factory hook
- [x] Enhance db-update factory hook
- [x] Remove deprecated global variables in favor of requests in hooks and settings
- [x] Update Acquia host forwarding and site path settings.php config in favor of requests

### PR DESCRIPTION
**Pull request by lcatlett**
Fixes #3158 subtasks
--------

Changes proposed:
---------
- Creates and sets unique drush cache directory per site process for post-install site tasks on ACSF
- Adds BLT update hook to refresh hooks

Steps to replicate the issue:
----------
the original issue can be replicated by running a site install / create new site on ACSF. When the existing post-install Factory hook is executed, race conditions result from long running drush tasks such as `cc drush` which then cause failures in importing config and bootstrapping the correct database.

Steps to verify the solution:
-----------
1. Deploy PR to ACSF environment
2. Create site
2. Validate that post-install hook executes without errors. To test via the command line as if the task was run by the factory, run `CACHE_PREFIX='/mnt/tmp/sandbox.01live' \drush8 --root='/mnt/www/html/sandbox.01live/docroot' -l 'site.sandbox.acsitefactory.com' scr --script-path='/mnt/www/html/sandbox.01live/factory-hooks/post-install' 'post-install.php'`, substituting your site, environment, database role, and uri as needed.

### CODE REVIEW

File: settings/blt.settings.php [hunk 1]
```diff
@@ -92,15 +92,11 @@
  * Site directory detection.
  */
 if (!isset($site_path)) {
-  try {
-    $site_path = DrupalKernel::findSitePath(Request::createFromGlobals());
-  }
-  catch (BadRequestHttpException $e) {
-    $site_path = 'sites/default';
-  }
+  $site_path = \Drupal::service('site.path');
```
Comment (ba66e77) [inline settings/blt.settings.php]: Why is this change necessary?

File: src/Update/Updates.php [hunk 2]
```diff
@@ -627,4 +627,26 @@ public function update_9002000() {
     $this->updater->getOutput()->writeln($formattedBlock);
     $this->updater->getOutput()->writeln("");
   }
+
+    /**
+   * 9.2.1.
```
Comment (ba66e77) [inline src/Update/Updates.php]: Should this be 9.2.1 since it's being applied to 9.2.0 and not 9.2.1 (presumably)?

File: settings/blt.settings.php (hunk 1 above)
Comment (lcatlett) [reply settings/blt.settings.php]: I inadvertently put the notes about this in #3172 rather than this one since the two PRs are dependent on one another: Update site path detection to use core site.path service due to findSitePath bug always returning first default directory (see https://www.drupal.org/project/drupal/issues/2280409).

File: src/Update/Updates.php (hunk 2 above)
Comment (lcatlett) [reply src/Update/Updates.php]: I deliberately made the hook 9.2.1 to ensure that the factory hook templates were refreshed for both post-install and db-update since db-update was merged previously in #3166. In looking back through the other hooks again, it doesn't seem like this has been a standard practice so I will remove the 9.2.1 hook

**Comment by egnikitin** [comment]
I'm sorry, but looks like this issue has not been resolved completely.
